### PR TITLE
V22F-109 Visual Recording

### PR DESCRIPTION
**Pull request by suppergerrie2**
Adds a button that allows the user to watch recordings back

### CODE REVIEW

**Review by noutheesink — APPROVED** [review]
Backend seems to be correct

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/FileDialogManager.cpp [hunk 1]
```diff
@@ -183,9 +184,16 @@ void Release(IShellItem* p_shellItem, IFileDialog* p_fileDialog)
 /// @param p_exts     The extension filters of the types of file to select
 ///	@param p_extCount The amount of extensions/names provided
 /// @note             See SelectBlackBox in ResearcherMenu.cpp for an example on how to call this function
-bool SelectFile(char* p_buf, char* p_err, bool p_folder, const wchar_t** p_names, const wchar_t** p_exts, int p_extCount)
+bool SelectFile(char* p_buf, char* p_err, bool p_folder, const wchar_t** p_names, const wchar_t** p_exts, int p_extCount, bool p_linuxDirectories)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/FileDialogManager.cpp:187]: Missing new param in documentation

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/FileDialogManager.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ void Release(IShellItem* p_shellItem, IFileDialog* p_fileDialog, COMDLG_FILTERSP
...
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/FileDialogManager.cpp:33]: missing new param in documentation

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 3]
```diff
@@ -407,6 +409,7 @@ TEST(RecorderTests, WriteRunSettingsTests)
     GfInit(false);                                                                          \
     filesystem::path varName;                                                               \
     filesystem::current_path(SD_DATADIR_SRC);                                               \
+    SetupSingletonsFolder();                                                                \
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:412]: ASSERT_TRUE for setupsingletonsfolder

**Review by larsdekwant — CHANGES_REQUESTED** [review]
- Some style changes
- Sometimes the game stops receiving data from the blackbox. if you pause the game, it is stuck on `IPC_OK(m_server.AwaitData(m_buffer, SBB_BUFFER_SIZE), "Failed to receive data");` in SocketBlackBox::GetDecisions() 
- This error is hard to reproduce, so i'm not sure what we will do with this? Maybe add a bug report for later? Seems like an IPCLib issue.

File: speed-dreams/source-2.2.3/data/config/raceman/replay_visual.xml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    <attstr name="name" val="Replay"/>
+    <attstr name="type" val="Replay"/>
+    <attstr name="description" val="Replay"/>
+    <attnum name="priority" val="90.000000"/>
+  </section>
+
+  <section name="Tracks">
+    <attnum name="maximum number" val="1.000000"/>
+    <section name="1">
+      <attstr name="name" val="test_highway"/>
```
[the remark this answers is not in the document]
Comment (suppergerrie2) [reply speed-dreams/source-2.2.3/data/config/raceman/replay_visual.xml:16]: It's overridden before it's used, it doesn't matter what's in it

**Review by larsdekwant — APPROVED** [review]

**Pull request merged**